### ISSUE TITLE
Creating a new file while sidebar is hidden could be improved

### ISSUE THREAD

**Issue by misolori** [microsoft/vscode#64434]
![gif](https://user-images.githubusercontent.com/35271042/49519667-cbf42b80-f8a1-11e8-8ce1-6c31712ef4f9.gif)

Steps to Reproduce:

1. Open file
2. Hide sidebar (Cmd+B)
3. Command Palette > New file
4. Sidebar appears with new file added next to the current file

It would be great if running `File: New File` from the command palette would show you a picker with the path pre-populated, allowing you to traverse through folder and also preserving the users' view. Here's a quick mockup:

<img width="824" alt="image" src="https://user-images.githubusercontent.com/35271042/49519711-ee864480-f8a1-11e8-8282-2acb0cd9a582.png">

cc @stevencl

**Comment by isidorn** [comment]
@misolori this makes sense to me. Would you be willing to provide a PR for this? Should be pretty straight forward. If yes, let me know and I can provide code pointers and guidance.

Also what would happen if the sidebar is open, but another viewlet is shown, not the explorer, for example debug? I think than the explorer should get opened and we keep the current experience.
Also what would you expect if a user enters an invalid file / folder name? Currently we show the error inline, and unfortunetly we do not have this support in quick pick.

Overall I think this change makes sense however I am not 100% convinced we need it. Can you elaborate where is the motivation coming from?
For example creating a new folder, I think the most common scenario is that you want to do something with it and that opening the explorer is useful. For file I see that you do not really need the explorer.

**Comment by misolori** [comment]
@isidorn I wouldn't mind taking a first stab it, code pointers would be great.

I think my proposed changes should only happen when the sidebar is collapsed. For the quickpick validation, I know I've seen this in other views (@chrmarti correct me if I'm wrong).

The main motivation is for the example I showed above where I'm trying to work from a "minimal" mode and am hiding the explorer intentionally but need to create a single file. While not absolutely necessary to this workflow, it's one of those "delights" where you hope the editor is smart enough to persist your view.

**Comment by chrmarti** [comment]
(We have validation for the input box, but not QuickPick. We could surface it for QuickPick too since much of the implementation is shared between the two.)

**Comment by isidorn** [comment]
@misolori agree, only when the sidebar is collapsed.
It would be great if you create a PR and than we can take it from there
Though please create the PR from [this branch](https://github.com/Microsoft/vscode/pull/65008), since it changes the whole explorer. Or alternativly wait for a week for us to merge in that branch to master. 

Here's the code pointer 
https://github.com/Microsoft/vscode/blob/e899bc638919ad4b11f2cd0e5915e1a79012fbf2/src/vs/workbench/parts/files/electron-browser/fileActions.ts#L1150
Basically that code path is only used for new file and new folder actions. Exactly what you need. You should use the `partService` to check if the sidebar is visible, if not do your thing vie `quickOpenService` and `fileService`.

Once we get that initial stuff working we can include @chrmarti to surface us validation for the QuickPick so you also adopt that.

**Comment by isidorn** [comment]
I do not plan to look into this this milestone. Leaving up to @misolori , assign back to me if you need help.

**Comment by misolori** [comment]
Not planning on tackling this for this month.

**Comment by misolori** [comment]
Would love it if someone could create a PR for this, see [code pointer above](https://github.com/microsoft/vscode/issues/64434#issuecomment-451484158).
